### ISSUE TITLE
Please, explain the phrase in the documentation better

### ISSUE THREAD

**Issue by Anton-Latukha** [aesiniath/unbeliever#49]
https://github.com/aesiniath/unbeliever/blob/d1ec2cf6093b30408fb08a0cdcbfe2f1cbab2ac1/core-program/lib/Core/Program/Execute.hs#L32

Probably what was meant, is simply without "when", resulting in: "`To avoid this, all output is sent through a single channel.`".

**Comment by istathar** [comment]
Hi Anton! It means, in effect, that there's no stderr. Whether you're doing output or "logging" it is written to stdout _ordered_. So if you write something with `write`, then a `debug` statement, and then something else with `write` the output will be in program order.

I'll see if I can make that clearer in the documentation.

**Comment by Anton-Latukha** [comment]
Yes.

`To avoid this, all output is sent through a single channel.` means single thread FIFO ordered. `when` word there is unnecessery - moreover it is harmful, it creates gramatically incorrect indecipherable sentence, because without `when` - thre is a ... -> To avoid this -> ... logical flow. With `when` it becomes `... -> To avoid this <- ...`, now logical structure converges on "To avoid this" and not gives logical continuation.

In short, read close why `when` word is there, does it fit how the logical flow.
